src/core/algorithms/dc/model/column_operand.h

```diff
@@ -3,79 +3,134 @@
 #include <cstdlib>
 #include <exception>
 #include <memory>
+#include <optional>
 #include <string>
 #include <vector>
 
-#include <boost/functional/hash.hpp>
 #include <easylogging++.h>
 
 #include "model/table/column.h"
 #include "model/table/relation_data.h"
+#include "model/types/type.h"
 
 namespace algos::dc {
 
+// kS and kT are used to tell which tuple is used in ColumnOperand.
+// To describe Predicate tuple also can be used kMixed since
+// ColumnOperand's may have different tuples.
+enum class Tuple { kS, kT, kMixed };
```

Comment:
Sorry, I don't understand the English here: "To describe Predicate tuple also can be used kMixed".
And still I don't understand what kMixed is. Suppose we have two attributes: A and B, and two tuples: T and S.
We can form different types of predicates: A.t op A.t, A.t op A.s, A.s op A.s, A.s op A.t, A.s op B.t, A.t op B.t and so on.. Also any of the column operand could be a constant.
What does A.mixed means? Can we form predicate A.mixed op B.t, for example?